src/blockrelay/graphene.h

```diff
@@ -144,13 +144,9 @@ class CGrapheneBlock
         if (!pGrapheneSet)
         {
             if (version > 3)
-                pGrapheneSet = std::make_shared<CGrapheneSet>(CGrapheneSet(3, computeOptimized));
-            else if (version == 3)
-                pGrapheneSet = std::make_shared<CGrapheneSet>(CGrapheneSet(2));
-            else if (version == 2)
-                pGrapheneSet = std::make_shared<CGrapheneSet>(CGrapheneSet(1));
+                pGrapheneSet = std::make_shared<CGrapheneSet>(CGrapheneSet(version - 1, computeOptimized));
             else
-                pGrapheneSet = std::make_shared<CGrapheneSet>(CGrapheneSet(0));
+                pGrapheneSet = std::make_shared<CGrapheneSet>(CGrapheneSet(version - 1));
```

Comment:
The version handling is starting to get a bit confusing (at least for me, but I've been admittedly out of the loop for a while).  Why is `version-1` used here and correct? Maybe just add a couple lines of docs to this?

I am also wondering whether versions should become enums maybe and warrant some more in-code documentation?

Reply:
Yes, agreed that versioning is getting complicated. I went ahead and added comments regarding the use of the version-1 idiom. Can we talk more offline about how to simplify and clarify graphene-related versions? Perhaps we can target those changes for a different PR?